### PR TITLE
Branch userguide update

### LINKED ISSUE #83: [PE-D][Tester B] edit command for tag does not follow behaviour mentioned in help message
the help message claims that '/' values can be taken in by -t, but this is not true, pretty much just  a minor oversight, but should be corrected so as to not confuse users

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: Ragnapop/ped#1

### LINKED ISSUE #86: [PE-D][Tester B] tag for add command can take in duplicate values
when keying in duplicate values for the tag value in add command, duplicate tags are allowed, but merge into one tag in the UI.
Low severity but may be slightly confused for new users.

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: Ragnapop/ped#2

### LINKED ISSUE #92: [PE-D][Tester A] Auto formatting for role
**Command Executed**

**Result**

The application will automatically convert the role into a specific format. I'm not sure if it's intended.

The main effect of this will be QA -> Qa, Power BI -> Power Bi, ML -> Ml, AI -> Ai

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: JustWeiHao/ped#5

### LINKED ISSUE #95: [PE-D][Tester B] no visual examples for sort command
the sort command may be quite confusing for new users as there are multiple ways to re-sort the order of the entries and I honestly feel that the documentation given may not be sufficient especially if the person has a lot of entries in their intern book.
This is just a recommendation but i would suggest putting some pictures of expected behavior into the user guide for clarity.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: Ragnapop/ped#4

### LINKED ISSUE #102: [PE-D][Tester B] visual bug in tags/tags are case insensitive
The tags input is case insensitive, so the tags list inputs that are all capitalized as only the first letter is uppercase. It may be helpful to change this or list this behaviour in the user guide as the user guide did not state that tags are insensitive to casing.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Ragnapop/ped#6

### LINKED ISSUE #109: [PE-D][Tester E] UG does not discuss the behavior of the optional field -d1 and -d2 if one of them is not present
The application has the behavior that not adding "-d2" ignores if "-d1" is here. This seems to be intended behavior and it makes sense. However, it is not stated in the User Guide.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: pjanthony2001/ped#5

### LINKED ISSUE #112: [PE-D][Tester E] Allows adding invalid phone numbers that are 4 digits long
The command entered was: add -n NUS -t SoftwareEngineer -e NUS@nus.edu.com -p 0987 -d1 2024-07-07 -d2 2024-08-08 

This allowed the phone number to be less than 8 digits, which is invalid (but this is debatable). However, this is not shown in the UG to be a constraint/not a constraint of the field for the user to be able to enter numbers that are less than 8 digits long. In the application, there is an error message for inputting 3 digit numbers or lower, but it should be in the UG as a constraint to the field

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: pjanthony2001/ped#6

### LINKED ISSUE #116: [PE-D][Tester C] Documentation does not include limitations of each prefix
Details: Will only get to know the limitations when shown the error message  trying out the command. For example, documentation not given to inform user that company name cannot include symbols. E.g. add -n @@@ -e example@gmail.com -t finance team intern
 

 user may be caught surprised by new limitations.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: ValenciaLim/ped#1

### LINKED ISSUE #121: [PE-D][Tester E] UG should contain constraints for email
There is no description of the email constraints in the UG, but there is in the application. It would be good if these were present in both, and the error message in the application is not too long so that the user feedback box is not over crowded

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: pjanthony2001/ped#8

### LINKED ISSUE #126: [PE-D][Tester E] Sorting sorts not only by alphabetical order but alphanumerical order which is not in the UG
I used the `sort a` command and was able to sort not only by alphabetical order. However, this is not stated in the UG where it says only alphabetical order. Its a minor nitpick, but does have relevance as your Company names can have numbers in them.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: pjanthony2001/ped#13

### LINKED ISSUE #129: [PE-D][Tester E] UG should contain constraints for dates
Command Entered: `add -n NUS -t SoftwareEngineer -e NUS@nus.edu.com -p 91234567 -d1 07-02-2022 -d2 2024-08-08`

The UG should contain constraints for the date format which would be consistent with the rest of the UG, similar to the constraints for `INDEX`

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: pjanthony2001/ped#16

### PR DESCRIPTION
**Pull request by Xilef121**
Provide more details regarding these functions, giving users better ideas of the constraints of parameters.

closes #129
closes #126 
closes #121 
closes #116
closes #112
closes #109
closes #102
closes  #95 
closes #92
closes #86
closes  #83

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -178,15 +168,23 @@ Adds a company to the InternBook.
...
+| End Date                   | Optional            | -d2  | Dates should be in the form YYYY-MM-DD<br/>End date must be later than start date                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
```
Comment (cocoanautz) [inline docs/UserGuide.md:178]: For dates, could mention that start date can be same as end date

File: docs/UserGuide.md [hunk 2]
```diff
@@ -178,15 +168,23 @@ Adds a company to the InternBook.
...
+| Email                      | Compulsory          | -e   | Emails should be of the format local-part@domain and adhere to the following constraints:<br/>1. The local-part should only contain alphanumeric characters and these special characters, excluding the parentheses, (+_.-). The local-part may not start or end with any special characters.<br/>2. This is followed by a '@' and then a domain name. The domain name is made up of domain labels separated by periods.<br/>The domain name must:<br/>- end with a domain label at least 2 characters long<br/>- have at least 2 domain labels separated by a dot (.)<br/>- have each domain label start and end with alphanumeric characters<br/>- have each domain label consist of alphanumeric characters, separated only by hyphens, if any. |
```
Comment (cocoanautz) [inline docs/UserGuide.md:174]: Might want to mention that local-part cannot contain consecutive special characters.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -204,6 +202,14 @@ Format: `edit INDEX [-n NAME] [-p PHONE_NUMBER] [-e EMAIL] [-t TAG…]​ [-d1 S
...
+| End Date                   | Optional            | -d2   | Dates should be in the form YYYY-MM-DD<br/>End date must be later than start date                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
```
Comment (cocoanautz) [inline docs/UserGuide.md:212]: Same as above

File: docs/UserGuide.md (hunk 1 above)
Comment (Xilef121) [reply docs/UserGuide.md]: made the suggested ammendments

File: docs/UserGuide.md (hunk 2 above)
Comment (Xilef121) [reply docs/UserGuide.md]: added

**Review by SherwynNg — COMMENTED** [review]
LGTM!

**Pull request merged**